setup.py

```diff
@@ -58,6 +59,16 @@
     ],
 }
 
+console_scripts = list(itertools.chain.from_iterable([
+    [
+        'sphinx-build%s = sphinx.cmd.build:main' % version,
+        'sphinx-quickstart%s = sphinx.cmd.quickstart:main' % version,
+        'sphinx-apidoc%s = sphinx.ext.apidoc:main' % version,
+        'sphinx-autogen%s = sphinx.ext.autosummary.generate:main' % version,
+    ] for version in ('', '-%s' % sys.version[:1], '-%s' % sys.version[:3])
```

Comment:
Why `version[:3]` is needed?

Reply:
This was taken from `pip`. I assume newer versions of Python3 add features that are not backwards compatible, like the addition of the [`async` keyword in Python3.4](https://docs.python.org/3/library/asyncio.html). As a result, a fully versioned binary would be helpful

Reply:
It might be useful if python has (or will have) breaking compatibility between minor version.
If not, this is not needed. So I don't feel this is needed.